connectors/sources/google_cloud_storage.py

```diff
@@ -203,16 +203,10 @@ async def validate_config(self):
         """Validates whether user inputs are valid or not for configuration field.
 
         Raises:
-            Exception: The service account json is empty.
             Exception: The format of service account json is invalid.
         """
-        if (
-            self.configuration["service_account_credentials"] == ""
-            or self.configuration["service_account_credentials"] is None
-        ):
-            raise ConfigurableFieldValueError(
-                "Google Cloud service account json can't be empty."
-            )
+        self.configuration.check_valid()
```

Comment:
Question: Why do we need `validate_config` and a manual call to `check_valid` sometimes? I suppose that we sometimes need more "advanced" validations, but do we always have to call `check_valid` manually then?

Reply:
Yeah I'd also rename current `validate_config` to something else to indicate that it's non-standard validation

Reply:
(sorry if anyone read that and the word "validation" lost all meaning)

Reply:
If we conceptualise these as separate validation checks, then I'm okay with splitting them up. So `check_valid()` covers framework-level, another method on the connector can cover connector-level.

I just checked impact and a lot of files will be updated. Is it okay to make that a followup PR after this?

Reply:
Definitely, also curious to hear @artem-shelkovnikov's opinion on that change as it'll be a change with a broader impact.

Reply:
Note: this will be followed up as a stretch task https://github.com/elastic/enterprise-search-team/issues/4302